langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -228,6 +273,15 @@ private WeaviateObject buildObject(String id, Embedding embedding, String text)
 
     private static EmbeddingMatch<TextSegment> toEmbeddingMatch(Map<String, ?> item) {
         Map<String, ?> additional = (Map<String, ?>) item.get(ADDITIONALS);
+        final Metadata metadata = new Metadata();
+        if (item.get(METADATA) != null && item.get(METADATA) instanceof Map) {
+            Map<String, ?> resultingMetadata = (Map<String, ?>) item.get(METADATA);
+            for(Map.Entry<String, ?> entry : resultingMetadata.entrySet()) {
+                if(entry.getValue() != null && !"".equals(entry.getValue())) {
```

Comment:
how to know if there was no metadata entry or there was, but with an empty string value?

Reply:
Yes, i used "null" instead which seems to be the correct value for null